contentcuration/contentcuration/management/commands/mark_incomplete.py

```diff
@@ -94,23 +85,36 @@ def handle(self, *args, **options):
         logging.info('Marked {} questionless exercises (finished in {})'.format(count, time.time() - exercisestart))
 
         exercisestart = time.time()
-        query = ContentNode.objects\
+
+        exercise_check_query = With(AssessmentItem.objects.exclude(type=exercises.PERSEUS_QUESTION)
+                                    .filter(
+                                        Q(question='') |
+                                        Q(answers='[]') |
+                                        # hack to check if no correct answers
+                                        (~Q(type=exercises.INPUT_QUESTION) & ~Q(answers__iregex=r'"correct":\s*true'))).order_by(), name="t_assessmentitem")
+
+        query = exercise_check_query.join(ContentNode, id=has_questions_query.col.contentnode_id, _join_type=LOUTER)\
+            .with_cte(exercise_check_query) \
+            .annotate(t_contentnode_id=exercise_check_query.col.contentnode_id) \
             .filter(kind_id=content_kinds.EXERCISE) \
             .exclude(complete=False) \
+            .filter(t_contentnode_id__isnull=False)\
```

Comment:
We can drop this filter if we remove `_join_type=LOUTER` as it would then default to an inner join, which should be the same

Reply:
Have updated accordingly, thanks!